lib/util.c

```diff
@@ -1460,6 +1460,61 @@ grn_inspect_limited(grn_ctx *ctx, grn_obj *buffer, grn_obj *obj)
   return buffer;
 }
 
+grn_obj *
+grn_inspect_key_with_table(grn_ctx *ctx, grn_obj *buffer, grn_obj *table, const void *key, unsigned int key_size)
+{
+  grn_obj inspected;
+
+  if (!buffer) {
+    buffer = grn_obj_open(ctx, GRN_BULK, 0, GRN_DB_TEXT);
+  }
+
+  if (!table) {
+    ERR(GRN_INVALID_ARGUMENT,
+        "failed to inspect key with table: table must not NULL");
+    return buffer;
+  }
+
+  if (table->header.type == GRN_TABLE_NO_KEY) {
+    GRN_TEXT_INIT(&inspected, 0);
+    grn_inspect_name(ctx, &inspected, table);
+    grn_text_printf(ctx, buffer, "no need to inspect key because table is no_key: <%.*s>",
+                    (int)GRN_TEXT_LEN(&inspected), GRN_TEXT_VALUE(&inspected));
+    GRN_OBJ_FIN(ctx, &inspected);
+    return buffer;
+  }
```

Comment:
This is different behavior from other `grn_inspect()` family.

`\#<key (null) table:\#<no_key #{table_name}>>` or something is better.

Reply:
I've fixed to follow .